openapi/cosmos.yaml

```diff
@@ -332,6 +332,144 @@ paths:
+        '503':
+          $ref: '#/components/responses/statusServiceUnavailable'
+        '504':
+          $ref: '#/components/responses/statusGatewayTimeout'
+
+  /mdr/specializations/sdtm/datasetspecializations:
+    get:
+      operationId: api.products.specializations.sdtm.get_datasetspecializations
+      description: Get SDTM Dataset Specialization List across Packages, optionally limited to a domain
+      tags:
+        - Study Data Tabulation Model Dataset Specializations (SDTM)
+      parameters:
+        - $ref: '#/components/parameters/sdtmDatasetSpecializationDomain'
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/sdtmDatasetSpecializationsLatest'
+        '400':
+          $ref: '#/components/responses/statusBadRequest'
+        '401':
+          $ref: '#/components/responses/statusUnauthorized'
+        '403':
+          $ref: '#/components/responses/statusForbidden'
+        '404':
+          $ref: '#/components/responses/statusNotFound'
+        '405':
+          $ref: '#/components/responses/statusNotAllowed'
+        '406':
+          $ref: '#/components/responses/statusNotAcceptable'
+        '500':
+          $ref: '#/components/responses/statusInternalError'
+        '503':
+          $ref: '#/components/responses/statusServiceUnavailable'
+        '504':
+          $ref: '#/components/responses/statusGatewayTimeout'
+
+  /mdr/specializations/sdtm/datasetspecializations/{datasetspecialization}:
+    get:
+      operationId: api.products.specializations.sdtm.get_datasetspecialization
+      description: Get SDTM Dataset Specialization (Latest)
+      tags:
+        - Study Data Tabulation Model Dataset Specializations (SDTM)
+      parameters:
+        - $ref: '#/components/parameters/sdtmDatasetSpecialization'
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/sdtmDatasetSpecializationLatest'
+        '400':
+          $ref: '#/components/responses/statusBadRequest'
+        '401':
+          $ref: '#/components/responses/statusUnauthorized'
+        '403':
+          $ref: '#/components/responses/statusForbidden'
+        '404':
+          $ref: '#/components/responses/statusNotFound'
+        '405':
+          $ref: '#/components/responses/statusNotAllowed'
+        '406':
+          $ref: '#/components/responses/statusNotAcceptable'
+        '500':
+          $ref: '#/components/responses/statusInternalError'
+        '503':
+          $ref: '#/components/responses/statusServiceUnavailable'
+        '504':
+          $ref: '#/components/responses/statusGatewayTimeout'
+
+################################################################################
+
+# Dataset Specialization Paths
+
+  /mdr/specializations/datasetspecializations/{biomedicalconcept}:
```

Comment:
Could this conflict if a specialization has the same code as a different biomedical concept? I think we should differentiate when we are searching for specializations by bc vs by specializations name

Reply:
Yes, that is a good point. They would never have the same code, but it is not clear from the endpoint whether the code is a BC or datasetspecialization. 
Would this work: /mdr/specializations/datasetspecializations/biomedicalconcept/{biomedicalconcept}
Or should we make biomedicalconcept into a query parameter?

Btw, note that this endpoint does not have 'sdtm' because when there are more dataset specializations in the future (sdtm, cdash, ...) they would all be returned.

Reply:
However, If we use a query param, there should be functionality if none is provided. Is that the case?

Reply:
That would make sense, but the URl would just be /mdr/specializations/datasetspecializations/ which would return all specializations. Then you could filter with the query  param so the base would return all specializations

Reply:
Understood. I meant either /mdr/specializations/datasetspecializations/biomedicalconcept/{biomedicalconcept} or /mdr/specializations/datasetspecializations with a query parameter.
I will make the change.